crates/shared/src/http_solver/model.rs

```diff
@@ -193,7 +193,7 @@ pub enum Score {
     /// To have more flexibility, the protocol score can be tweaked by the
     /// solver by providing a multiplication factor. Expected value: [0,
     /// inf], 0 being 100% discount, 1 being the same as protocol score
-    MulFactor(f64),
+    ScoreFactor(f64),
```

Comment:
Is there an advantage of using a scoreFactor (as in this PR) over a scoreDiscount as (proposed in the text of the CIP)?
If I understand the code correctly you scale the full objective value and not just the quality. If this is true than a scoreFactor and a scoreDiscount are not equivalent.
An advantage of the latter is that you can be (to some extend) oblivious to transaction costs: If cost estimates by us were accurate and a submitting solver does not know about costs of their solution, then an additive discount can be used to include revert risk into the score. With a multiplicative scaling of the objective you cannot easily do this. You would have to scale the quality only.

Thus I would propose to either use a `scoreDiscount` as in the proposal or to only scale the quality (surplus + fee) instead of the whole objective (surplus + fee - cost).

Reply:
I thought it might be easier for solvers to start with the factor, because it is scaled together with the objective_value. But I do agree we should implement as defined in the CIP (if I had any suggestions I should have brought it during the CIP discussion).
Fixed.